browser_history/generic.py

```diff
@@ -33,59 +34,82 @@ class Browser:
+      exist. At least one of the following must be set:
+      :py:class:`windows_path`, :py:class:`mac_path`, :py:class:`linux_path`
+    * :py:class:`history_file`
+    * :py:class:`history_SQL`
 
-    name = "Generic"
+    These following class variable can optionally be set:
 
-    windows_path = None
-    mac_path = None
-    linux_path = None
+    * :py:class:`bookmarks_file`
+    * :py:class:`bookmarks_parser`
+    * :py:class:`profile_dir_prefixes`
+    * :py:class:`profile_support`
+    * :py:class:`_local_tz`
 
-    profile_support = False
-    profile_dir_prefixes = []
+    :param plat: the current platform. A value of :py:class:`None` means the platform
+       will be inferred from the system.
 
-    history_file = None
-    bookmarks_file = None
+    Examples:
 
-    history_SQL = None
+    >>> class CustomBrowser(Browser):
+    ...     linux_path = 'browser'
+    ... vars(CustomBrowser())
+    {'profile_dir_prefixes': [], 'history_dir': PosixPath('/home/username/browser')}
+    """
 
-    bookmarks_parser = lambda bookmark_path: None  # noqa: E731
+    windows_path: typing.Optional[str] = None  #: browser path on Windows.
+    mac_path: typing.Optional[str] = None  #: browser path on Mac OS.
+    linux_path: typing.Optional[str] = None  #: browser path on Linux.
 
-    _local_tz = datetime.datetime.now().astimezone().tzinfo
+    profile_support: bool = False  #: boolean indicating whether
+    """Boolean indicating whether the browser supports multiple profiles."""
 
-    def __init__(self, plat: utils.Platform = None):
+    profile_dir_prefixes: typing.Optional[typing.List[typing.Any]] = None
+    """List of possible prefixes for the profile directories.
+    Keep empty to check all subdirectories in the browser path.
+    """
+
+    bookmarks_file: typing.Optional[str] = None
+    """Name of the (SQLite, JSON or PLIST) file which stores the bookmarks."""
+
+    bookmarks_parser: typing.Callable = lambda bookmark_path: None
+    """A function to parse bookmarks and convert to readable format."""
+
+    _local_tz: typing.Optional[datetime.tzinfo] = (
+        datetime.datetime.now().astimezone().tzinfo
+    )
+    """Gets a datetime object of the current time as per the users timezone."""
+
+    history_dir: Path
+    """History directory."""
+
+    @property
+    @abc.abstractmethod
+    def name(self) -> str:
+        """A name for the browser. Not used anywhere except for logging and errors."""
+
+    @property
+    @abc.abstractmethod
+    def history_file(self) -> str:
+        """Name of the (SQLite) file which stores the history."""
+
+    @property
+    @abc.abstractmethod
+    def history_SQL(self) -> str:
+        """SQL query required to extract history from the ``history_file``.
+        The query must return two columns: ``visit_time`` and ``url``.
+        The ``visit_time`` must be processed using the
+        `datetime <https://www.sqlitetutorial.net/sqlite-date-functions/sqlite-datetime-function/>`_  # noqa: E501
```

Comment:
The `# noqa: E501` shows up in the docs xD
![image](https://user-images.githubusercontent.com/34161949/97568678-0ffbf680-1a0d-11eb-90a8-8e810968974d.png)